lib/ansible/modules/cloud/scaleway/scaleway_compute.py

```diff
@@ -0,0 +1,745 @@
+
+    if not response.ok:
+        msg = 'Error during server search: (%s) %s' % (response.status_code, response.json)
+        compute_api.module.fail_json(msg=msg)
+
+    search_results = response.json["servers"]
+
+    return search_results
+
+
+def server_attributes_should_be_changed(compute_api, target_server, wished_server):
+    try:
+        # return any([
+        #    target_server["ipv6"] != wished_server["ipv6"],
+        #    target_server["tags"] != wished_server["tags"]
+        # ])
+        return False
+    except KeyError:
+        compute_api.module.fail_json(msg="Error while checking if attributes should be changed")
+
+
+def server_change_attributes():
+    # TODO: We can use a PATCH HTTP call to patch a server with the relevant information we need
+    # TODO: check validate_patch for the complete list in api-compute
+    # TODO: we can modify only
+    # name: works
+    # tags: works
+    # enable_ipv6: Works
+    # dynamic_ip_required: Works
+    # bootscript: Works
+    # boot_type: Works
+    # volumes: Works
+    # security_group: Works
+
+    # extra_networks
+    # Example: {"extra_networks": ["foobar"]}
+
+    # state_detail: The patch goes fine (200) but nothing is changed.
+    # Example: {"state_detail": "pouet"}
+
+    # ipv6: The patch goes fine (200) but nothing is changed. (No check upon types)
+    # Example: {"ipv6": {"gateway": "", "address": "", "netmask": 23}}
+    # {"ipv6": {"gadseway": "", "address": "", "netmask": 23}}
+
+    # location: The patch goes fine (200) but nothing is changed. (No check upon types)
+    # Example: {"location": {"plarf": 34}}
+    pass
+
+
+def core(module):
+    api_token = module.params['oauth_token']
+    region = module.params["region"]
+    wished_server = {
+        "state": module.params["state"],
+        "image": module.params["image"],
+        "name": module.params["name"],
+        "commercial_type": module.params["commercial_type"],
+        "enable_ipv6": module.params["enable_ipv6"],
+        "tags": module.params["tags"],
+        "organization": module.params["organization"]
+    }
+
+    compute_api = ScalewayComputeAPI(module=module,
+                                     headers={'X-Auth-Token': api_token,
+                                              'Content-type': 'application/json'},
+                                     base_url=SCALEWAY_LOCATION[region]["api_endpoint"])
+
+    changed, summary = state_strategy[wished_server["state"]](compute_api=compute_api, wished_server=wished_server)
+    module.exit_json(changed=changed, msg=summary)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            oauth_token=dict(
+                no_log=True,
+                # Support environment variable for Scaleway OAuth Token
+                fallback=(env_fallback, ['SCW_TOKEN', 'SCW_API_KEY', 'SCW_OAUTH_TOKEN']),
+                required=True,
+            ),
+            image=dict(required=True),
+            name=dict(),
```

Comment:
In the module documentation `name` is required.

Reply:
A default name will be provided if a name is missing, therefore, it's not a required parameter. I've fixed the documentation.